docs/api.md

```diff
@@ -554,6 +563,184 @@ Response:
 }
 ```
 
+### `GET /api/saved_pages`
+
+Returns all a user's saved pages
+
+#### Request headers
+
+- `GOVUK-Account-Session`
+  - the user's session identifier
+
+#### JSON response fields
+
+- `govuk_account_session` *(optional)*
+  - a new session identifier
+- `saved_pages`
+  - an array of pages the user has saved, identified by their page path
+
+#### Response codes
+
+- 401 if the session identifier is invalid
+- 200 otherwise
+
+#### Example request / response
+
+Request (with gds-api-adapters):
+
+```ruby
+GdsApi.saved_page_api.get_saved_pages(
+    govuk_account_session: "session-identifier",
+)
+```
+
+Response when no pages are saved:
+
+```json
+{
+    "govuk_account_session": "YWNjZXNzLXRva2Vu.cmVmcmVzaC10b2tlbg==",
+    "saved_pages": []
+}
+```
+
+Response when a user has saved two pages:
+
+```json
+{
+    "govuk_account_session": "YWNjZXNzLXRva2Vu.cmVmcmVzaC10b2tlbg==",
+    "saved_pages": [
+      { "page_path": "/government/organisations/government-digital-service" },
+      { "page_path": "/government/organisations/cabinet-office" },
+  ]
+}
+```
+
+### `PUT /api/saved_pages/:page_path`
+
+Upsert a saved page in a user's account
+
+#### Request headers
+
+- `GOVUK-Account-Session`
+  - the user's session identifier
+
+#### Request parameters
+
+- `page_path`
+  - An escaped URL safe string that identifies the path of a saved page.
+
+#### JSON response fields
+
+- `govuk_account_session` *(optional)*
+  - a new session identifier
+- `saved_page`
+  - an object containing the page path of the successfully saved page
+
+#### Response codes
+
+- 401 if the session identifier is invalid
+- 422 if the page could not be saved (see [error: page cannot be saved](#page-cannot-be-saved))
+- 200 otherwise
+
+#### Example request / response
+
+Request (with gds-api-adapters):
+
+```ruby
+GdsApi.saved_page_api.save_page(
+    page_path: "/guidance/foo",
+    govuk_account_session: "session-identifier",
+)
+```
+
+```json
+{
+    "govuk_account_session": "YWNjZXNzLXRva2Vu.cmVmcmVzaC10b2tlbg==",
+    "saved_page": {
+      "page_path": "/guidance/foo"
+    },
+}
+```
+
+### `DELETE /api/saved_pages/:page_path`
+
+Remove a saved page from a user's account
+
+#### Request headers
+
+- `GOVUK-Account-Session`
+  - the user's session identifier
+
+#### Request parameters
+
+- `page_path`
+  - An escaped URL safe string that identifies the path of a saved page.
+
+#### Response codes
+
+- 401 if the session identifier is invalid
+- 404 cannot find a page with the provided path
+- 204 successfully deleted
+
+#### Example request / response
+
+Request (with gds-api-adapters):
+
+```ruby
+GdsApi.delete_saved_page.delete_saved_page(
```

Comment:
This says `GdsApi.delete_saved_page.delete_saved_page`, but should be `GdsApi. saved_page_api.delete_saved_page` - though, see my comment in gds-api-adapters about the separate class

Reply:
Done - yeah that was just copy pasta, It's a bit strange doing this bit of the docs, perhaps this should have been a follow up commit.

I figured I'd do the GDS API adaptors then come back and check these matched. For all I know we could rename a bunch of those in the review over there